src/worker.ts

```diff
@@ -0,0 +1,45 @@
+import BN from 'bn.js'
+import workerpool from 'workerpool'
+import { OrderbookFetcher } from './orderbook_fetcher'
+import { sortOrderbookBySerializedPrice } from './utilities'
+import { transitiveOrderbook, Fraction } from '@gnosis.pm/dex-contracts'
+
+function markets(orderbooksObject: object, base: string, quote: string, hops: number) {
+  const orderbooks = OrderbookFetcher.deserializeOrderbooks(orderbooksObject)
+  const transitive = transitiveOrderbook(orderbooks, base, quote, hops)
+  const result = sortOrderbookBySerializedPrice(transitive)
+  return result
+}
+
+function estimatedBuyAmount(orderbooksObject: object, base: string, quote: string, hops: number, quoteAmount: string, priceRoundingBuffer: number) {
+  console.log(base, quote, hops, quoteAmount, priceRoundingBuffer)
+  const orderbooks = OrderbookFetcher.deserializeOrderbooks(orderbooksObject)
+  const transitive = transitiveOrderbook(orderbooks, base, quote, hops)
+  const sellAmount = new BN(quoteAmount)
+  // The orderbook API only allows us to compute a price for selling base tokens. Here, we want to sell quote tokens.
+  // Thus, we invert the orderbook, sell the requested amount (now in base) and multiply the computed price with the
+  // sell amount in order to compute the buy amount in the original base token. A small rounding buffer is considered
+  // to increase the chances of matching the order if placed.
+  const estimatedPrice = transitive.inverted().priceToSellBaseToken(new BN(sellAmount))
+  console.log(estimatedPrice)
+  if (estimatedPrice !== undefined) {
+    const buyAmountInBase =
+      (1 - priceRoundingBuffer) * new Fraction(sellAmount, 1).mul(estimatedPrice).toNumber()
+    console.log(buyAmountInBase)
+    return {
+      baseTokenId: base,
+      quoteTokenId: quote,
+      buyAmountInBase: buyAmountInBase,
+      sellAmountInQuote: sellAmount.toNumber(),
+    }
+  } else {
+    // TODO: in case there is no result (estimedPrice == null) what should we return?
+    // I think previously it finished the request with req.end() without setting anything but now
+    // we set an empty object. What is more correct?
+    return {}
+  }
```

Comment:
TODO

Reply:
Actually I think it would be correct to set buyAmountInBase to 0. Either the token does not exist or there is not enough liquidity, in both cases you are expected to get 0 of the tokens you want to buy. This would also normalize the return types more.